src/index/metadata.rs

```diff
@@ -203,6 +205,31 @@ fn read_flac(path: &Path) -> Result<SongTags> {
 	})
 }
 
+#[cfg_attr(feature = "profile-index", flame)]
+fn read_mp4(path: &Path) -> Result<SongTags> {
+	let tag = mp4ameta::Tag::read_from_path(path)?;
+	let mut tags = SongTags {
+		artist: None,
+		album_artist: None,
+		album: None,
+		title: None,
+		duration: None,
+		disc_number: None,
+		track_number: None,
+		year: None,
+	};
+	tags.artist = Some(tag.artist().unwrap().to_string());
+	tags.album_artist = Some(tag.album_artist().unwrap().to_string());
+	tags.album = Some(tag.album().unwrap().to_string());
+	tags.title = Some(tag.title().unwrap().to_string());
+	tags.duration = Some(tag.duration().unwrap() as u32);
+	tags.disc_number = tag.disk_number().0.and_then(|d| Some(d as u32));
+	tags.track_number = tag.track_number().0.and_then(|d| Some(d as u32));
+	tags.year = tag.year().unwrap().parse::<i32>().ok();
```

Comment:
All the fields in `SongTags` are `Option<>` so it's fine to leave them as `None` when mp4ameta didn't return anything. Can you avoid using `unwrap` so that Polaris doesn't terminate when fields are missing?

Reply:
Done.